doc/src/devdocs/pkgimg.md

```diff
@@ -33,8 +33,22 @@ Dynamic libraries on macOS need to link against `-lSystem`. On recent macOS vers
 To that effect we link with `-undefined dynamic_lookup`.
 
 ## [Package images optimized for multiple microarchitectures](@id pkgimgs-multi-versioning)
-Similar to [multi-versioning](@ref sysimg-multi-versioning) for system images, package images support multi-versioning. If you are in a heterogeneous environment, with a unified cache,
-you can set the environment variable `JULIA_CPU_TARGET=generic` to multi-version the object caches.
+
+Similar to [multi-versioning](@ref sysimg-multi-versioning) for system images, package images support multi-versioning. This allows creating package caches that can run efficiently on different CPU architectures within the same environment.
+
+### Usage and constraints
+
+Package images can only target the same or more specific CPU features than their base system image. This constraint ensures compatibility and prevents runtime errors.
+
+To enable multi-versioning for package images, set the [`JULIA_CPU_TARGET`](@ref JULIA_CPU_TARGET) environment variable when building packages:
+
+```bash
+# Create multi-versioned package images for generic and optimized targets
+export JULIA_CPU_TARGET="generic;haswell"
+
+# For heterogeneous environments, use generic to ensure broad compatibility
+export JULIA_CPU_TARGET="generic"
```

Comment:
I'm not sure we should recommend `JULIA_CPU_TARGET="generic"` for heterogeneous systems, stuff like `JULIA_CPU_TARGET="generic;haswell"` is better _and_ provides the safe fallback of `generic`

Reply:
Maybe we should have an option that copies the value that the sysimage was built with?

Reply:
With https://github.com/JuliaLang/julia/pull/58970 in I've simplified this PR.